src/common/SelectionList.js

```diff
@@ -0,0 +1,176 @@
+  };
+
+  render() {
+    const { label, options, selectedKey } = this.props.navigation.state.params;
+
+    const optionsList = Object.keys(options).map(key => ({
+      key,
+      label: options[key].label,
+    }));
+
+    return (
+      <Screen title={label}>
+        <FlatList
+          ItemSeparatorComponent={OptionDivider}
+          data={optionsList}
+          keyExtractor={item => `${item.key}`}
+          renderItem={({ item, index }) => (
+            <Touchable onPress={() => this.onOptionSelect(item.key)}>
+              <View style={this.styles.item}>
+                <Label
+                  style={selectedKey === item.key && this.styles.selectedValue}
+                  text={item.label}
+                />
+                {selectedKey === item.key && <IconDone size={16} color={BRAND_COLOR} />}
+              </View>
+            </Touchable>
+          )}
+        />
+      </Screen>
+    );
+  }
+}
+
+type SelectorOuterProps = $ReadOnly<{|
+  label: string,
+  options: OptionsType,
+  selectedKey: string,
+  style?: ViewStyleProp,
+  onOptionSelect: (key: string) => void,
+|}>;
+
+type SelectorProps = $ReadOnly<{|
+  ...SelectorOuterProps,
+
+  dispatch: Dispatch,
+|}>;
+
+/**
+ * An selector component, which shows the the label and selected value
+ *
+ * @prop [style] - Style object applied to the outermost component.
+ * @prop label - label for the selection list.
+ * @prop options - all available options to select from.
+ * @prop selectedKey - key of currently selected option.
+ * @prop onOptionSelect - callback when any option is selected.
+ */
+export class Selector extends PureComponent<SelectorProps> {
+  static contextType = ThemeContext;
+  context: ThemeColors;
+
+  styles = {
+    wrapper: {
+      marginTop: 4,
+    },
+    row: {
+      flex: 1,
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+    },
+    value: {
+      color: 'gray',
+      marginTop: 4,
+    },
+    icon: {
+      alignSelf: 'center',
+    },
+    label: {
+      fontWeight: '400',
+    },
+  };
+
+  navigateToOptionListScreen = () => {
+    const { dispatch, options, label, selectedKey, onOptionSelect } = this.props;
+    dispatch(navigateToSelectionListOptionsScreen(label, options, selectedKey, onOptionSelect));
+  };
+
+  render() {
+    const { label, options, selectedKey, style } = this.props;
+
+    return (
+      <Touchable style={[style, this.styles.wrapper]} onPress={this.navigateToOptionListScreen}>
+        <View style={this.styles.row}>
+          <View>
+            <Label style={this.styles.label} text={label} />
+            <Label style={this.styles.value} text={options[selectedKey].label} />
+          </View>
+          <IconRight size={24} style={[{ color: this.context.color }, this.styles.icon]} />
+        </View>
+      </Touchable>
+    );
+  }
+}
+
+const SelectionScreen: {
+  Selector: ComponentType<SelectorOuterProps>,
+  OptionsScreen: ComponentType<OptionsScreenOuterProps>,
+} = {
```

Comment:
Hmm, when you have to write a cast like this, that's a sign something's gone wrong with our type-checking :slightly_smiling_face: 

It looks like no-argument `connect` falls into an `empty`-type trap. The result is that you could write something quite untrue for the props types in this cast, and Flow would accept it.

I just experimented a bit, and found a workaround that avoids having to repeat ourselves, and gets Flow successfully type-checking the props we pass in when we use the component. Write the `connect` call like so:
```
connect<{||}, _, _>()
```

So e.g.
```js
export default {
  Selector: connect<{||}, _, _>()(Selector),
  OptionsScreen: connect<{||}, _, _>()(OptionsScreen),
};
```

I'll go fix that in the existing places we have no-argument `connect` in the tree, too.

Reply:
> I'll go fix that in the existing places we have no-argument `connect` in the tree, too.

Done: 1e0006e5b